### PR TITLE
feat(website): update website codeblocks design

### PR DESCRIPTION
**Pull request by TheSisb**
This change improves the codeblocks on our website by starting the code portion closed, adding a view/hide code button, and adding a copy button. This also improves the general styling.

### CODE REVIEW

File: packages/paste-website/src/components/site-wrapper/sidebar/SidebarAnchor.tsx [hunk 1]
```diff
@@ -9,7 +9,9 @@ interface SidebarAnchorProps {
   onClick?: () => void;
 }
 
-const StyledSidebarAnchor = styled(Link)<SidebarAnchorProps>((props) =>
+const StyledSidebarAnchor = styled(Link, {
+  shouldForwardProp: (prop) => prop !== 'nested',
+})<SidebarAnchorProps>((props) =>
```
Comment (gloriliale) [inline packages/paste-website/src/components/site-wrapper/sidebar/SidebarAnchor.tsx:14]: Is this related to Codeblocks or a fix for something else?

File: packages/paste-website/src/components/shortcodes/live-preview/index.tsx [hunk 2]
```diff
@@ -1,64 +1,113 @@
...
-  scope,
-  disabled = false,
-  noInline = false,
-}) => {
+const LivePreview: React.FC<LivePreviewProps> = ({children, language = 'jsx', scope}) => {
+  const [viewCode, setViewCode] = React.useState(false);
+  const clipboard = useClipboard({copiedTimeout: 1200});
   const pasteTheme = useTheme();
+
+  const handleCopy = React.useCallback(() => {
```
Comment (gloriliale) [inline packages/paste-website/src/components/shortcodes/live-preview/index.tsx:26]: For my own understanding, why is it useful to do useCallback() here specifically?

**Review by gloriliale — APPROVED** [review]
This looks soooo awesome! Like I said on an earlier thread, this might just be the best way I've seen code previews be done after looking at many many design system docs for ideas. 

I left a few questions for my own understanding, and only have one issue that is really more of a dark mode tokens issue, so I'll bring that up w/ design. Basically, in dark mode, background-color-body and background-color-body-inverse point to the same alias. Those tokens are used here a lot, so the dark mode codeblocks don't have as much contrast as the light mode ones. 
Screen Shot 2021-11-30 at 10 25 56 AM
Not something for finagling in this PR tho!

File: packages/paste-website/src/components/shortcodes/live-preview/index.tsx [hunk 3]
```diff
@@ -1,64 +1,113 @@
...
+                  <>
+                    <ShowIcon decorative />
+                    <span>View code</span>
+                  </>
+                )}
+              </Button>
+            </Box>
+          )}
+          {isSmallCodeVariant || viewCode ? (
+            <Box position="absolute" bottom="space40" right="space40">
```
Comment (shleewhite) [inline packages/paste-website/src/components/shortcodes/live-preview/index.tsx:95]: To make sure screen reader users know that the text of the button changed, this copy/copied button should match the one on icons list page: https://github.com/twilio-labs/paste/blob/1b4111d4a47ea3af35a26605d8174d4290a04f4e/packages/paste-website/src/components/icons-list/IconCard.tsx#L53

maybe it should be its own component...

**Review by shleewhite — CHANGES_REQUESTED** [review]

**Comment by nkrantz** [comment]
Alert Dialog, Combobox, Data Grid, Date Picker, Disclosure, Form Pill, Label, Menu, Modal, Pagination, and others (I stopped checking) have certain example(s) that are throwing an error (unexpected token) and the examples themselves aren't rendering - just the code. Other than that, looks _really_ great so far.

File: packages/paste-website/src/components/site-wrapper/sidebar/SidebarAnchor.tsx (hunk 1 above)
Comment (TheSisb) [reply packages/paste-website/src/components/site-wrapper/sidebar/SidebarAnchor.tsx:14]: This is a fix for something else. I noticed that the console was spammed with a warning about a "nested" prop being passed to a DOM element. This fixes that.

File: packages/paste-website/src/components/shortcodes/live-preview/index.tsx (hunk 2 above)
Comment (TheSisb) [reply packages/paste-website/src/components/shortcodes/live-preview/index.tsx:26]: The library this uses recommends that approach.

**Comment by katieporter** [comment]
This looks so, so good so far! Just one q: I noticed that the examples are no longer editable - was that an intentional change? I don't have strong feelings about it one way or the other because I don't have a sense for how useful being able to edit the code is for our consumers, but I wanted to check.

File: packages/paste-website/src/components/shortcodes/live-preview/index.tsx [hunk 4]
```diff
@@ -1,64 +1,113 @@
...
+          {/* corrects an inbuilt 10px margin style so we can accurately use token values */}
+          <LiveEditor
+            style={{
+              margin: '-10px',
+              fontFamily: pasteTheme.fonts.fontFamilyCode,
+              backgroundColor: pasteTheme.backgroundColors.colorBackgroundBodyInverse,
+            }}
+          />
+          {isSmallCodeVariant || viewCode ? null : <CodeBlockOverlayShadow />}
+          {isSmallCodeVariant ? null : (
```
[the remark this answers is not in the document]
Comment (TheSisb) [reply packages/paste-website/src/components/shortcodes/live-preview/index.tsx:80]: Fixed.

File: packages/paste-website/src/components/shortcodes/live-preview/index.tsx [hunk 5]
```diff
@@ -1,64 +1,113 @@
...
+            <Box position="absolute" top="space40" right="space40">
+              <Button variant="secondary" size="small" onClick={handleToggleCodeEditor}>
```
[the remark this answers is not in the document]
Comment (TheSisb) [reply packages/paste-website/src/components/shortcodes/live-preview/index.tsx]: Good observation. I believe I fixed this now. Lmk if you see any other improvements, and thanks!

File: packages/paste-website/src/components/shortcodes/live-preview/index.tsx (hunk 3 above)
Comment (TheSisb) [reply packages/paste-website/src/components/shortcodes/live-preview/index.tsx:115]: Done.

**Comment by TheSisb** [comment]
> Alert Dialog, Combobox, Data Grid, Date Picker, Disclosure, Form Pill, Label, Menu, Modal, Pagination, and others (I stopped checking) have certain example(s) that are throwing an error (unexpected token) and the examples themselves aren't rendering - just the code. Other than that, looks _really_ great so far.

Woops, yeah I broke this subtly. Fixed now.

> This looks so, so good! Just one q: I noticed that the examples are no longer editable - was that an intentional change? I don't have strong feelings about it one way or the other because I don't have a sense for how useful being able to edit the code is for our consumers, but I wanted to check.

I had disabled editing when I still had the Open In Codesandbox button, and forgot to re-enable it when I cut that feature. I re-enabled editing for now until we re-introduce that feature. Good catch.

File: packages/paste-website/src/components/site-wrapper/sidebar/SidebarAnchor.tsx (hunk 1 above)
Comment (SiTaggart) [reply packages/paste-website/src/components/site-wrapper/sidebar/SidebarAnchor.tsx:14]: OMG I've been looking for this for months! 🙏🏼

**Review by SiTaggart — APPROVED** [review]

**Comment by katieporter** [comment]
One teeny tiny thing that I noticed—when the examples are wider than their container and become scrollable, they lose their right padding within the container, like so:
image

Is that possible to fix? It's _suuuper_ minor, so totally fine if that's a pain to maintain that padding in that state. Definitely not a blocker for this.

Everything else looks perfect!

**Review by shleewhite — APPROVED** [review]
Just one nit about the copy button. Looks great!

**Comment by TheSisb** [comment]
@katieporter I couldn't figure it out at this time unfortunately.

**Pull request merged**